gradle/shadow.gradle

```diff
@@ -0,0 +1,90 @@
+import org.gradle.api.internal.file.FileResolver
+
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import com.github.jengelman.gradle.plugins.shadow.tasks.DefaultInheritManifest
+import org.gradle.api.internal.file.FileResolver
+
+buildscript {
+    repositories { jcenter() }
+    dependencies {
+        classpath 'com.github.jengelman.gradle.plugins:shadow:1.2.4'
+    }
+}
+
+configure([project(':bson'), project(':mongo-java-driver')]) {
+    apply plugin: 'osgi'
+    apply plugin: 'com.github.johnrengelman.shadow'
+
+    configurations {
+        compile.extendsFrom shadow
+    }
+
+    shadowJar {
+        from sourceSets.main.output
+        configurations = [project.configurations.compile]
+        mergeServiceFiles()
+        relocate 'com.fasterxml', 'org.bson.shaded.fasterxml'
```

Comment:
Despite this line, shadowJar adds this to the manifest:

    Class-Path: classmate-1.3.1.jar

I don't know if it matters, but it seems incorrect given that we're shading this library

I also see a META-INF/maven/com.fasterxml/classmate directory, containing the classmate pom file, in the bson and mongo-java-driver jar files, and I'm not clear whether it needs to be there.

There's also a NOTICE and LICENSE file from classmate next to our MANIFEST.MF.  If there's going to be a license there it should be ours.